sw/applications/example_spi_write/main.c

```diff
@@ -25,13 +25,13 @@
 
 #if TARGET_SIM && PRINTF_IN_SIM
     #define PRINTF(fmt, ...)    printf(fmt, ## __VA_ARGS__)
-#elif TARGET_PYNQ_Z2 && PRINTF_IN_FPGA
+#elif (TARGET_PYNQ_Z2 || TARGET_ZCU104 || TARGET_NEXYS_A7_100T) && PRINTF_IN_FPGA
```

Comment:
In order to be a bit more general and not to have to add everytime a 'new' board define

#if TARGET_SIM
    #if PRINTF_IN_SIM
        #define PRINTF(fmt, ...)    printf(fmt, ## __VA_ARGS__)
    #else
        #define PRINTF(fmt, ...)    // No-op
    #endif
#elif PRINTF_IN_FPGA
    #define PRINTF(fmt, ...)    printf(fmt, ## __VA_ARGS__)
#else
    #define PRINTF(fmt, ...)    // No-op
#endif

Reply:
Done for the applications